xyz-psql-connector/src/main/java/com/here/xyz/psql/PSQLXyzConnector.java

```diff
@@ -129,38 +129,73 @@ protected XyzResponse processGetFeaturesByBBoxEvent(GetFeaturesByBBoxEvent event
 
       final BBox bbox = event.getBbox();
 
-      if(event.getTweakType() != null)
-      { Map<String, Object> tweakParams = event.getTweakParams();
+      boolean bTweaks = ( event.getTweakType() != null ),
+              bOptViz = "viz".equals( event.getOptimizationMode() );
 
-        switch (event.getTweakType().toLowerCase()) {
+      if( !bOptViz && event.getSelection() != null && "*".equals( event.getSelection().get(0) ) ) //correction for raw usecase
```

Comment:
"!bOptViz && "can be removed because when mode is "viz", tweak type will be ENSURE and then the setSelection is executed again on line 171. This means setSelection is always executed either when bOptViz is true or false.
In this case, line 135&136 is the only ones needed for setSelection logic. You can also remove lines 171&172 and adjust line 173 (remove "else ")

Reply:
this is in case of bTweaks=false, then the selection must be reseted to null in order to gain standard behaviour.